File: Game/src/main/java/net/winternetwork/bedwars/game/modules/generators/object/Generator.java
package net.winternetwork.bedwars.game.modules.generators.object;

import com.gmail.filoghost.holographicdisplays.api.Hologram;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.World;
import org.bukkit.block.Block;
import org.bukkit.configuration.ConfigurationSection;

@Getter
@ToString
@RequiredArgsConstructor
public class Generator {

    private final String name;

    private final Material itemType;

    @Setter
    private Block block;

    @Setter
    private int time = 10;

    @Setter
    private Hologram hologram;

    public static Generator fromSection(ConfigurationSection section) {

        Generator generator = new Generator(
                section.getName(),
                Material.matchMaterial(section.getString("drop_material"))
        );

        ConfigurationSection blockSection = section.getConfigurationSection("block");
        World world = Bukkit.getWorld(blockSection.getString("world"));

        generator.setBlock(
                world.getBlockAt(
                        blockSection.getInt("x"),
                        blockSection.getInt("y"),
                        blockSection.getInt("z")
                )
        );

        return generator;
    }
}

File: Game/src/main/kotlin/br/redebaiana/baianowars/modules/generators/model/Generator.kt
package br.redebaiana.baianowars.modules.generators.model

import com.gmail.filoghost.holographicdisplays.api.Hologram
import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.block.Block
import org.bukkit.configuration.ConfigurationSection

data class Generator(
    val name: String,
    val type: Material
) {

    lateinit var block: Block
    var time = 10

    lateinit var hologram: Hologram

    companion object {
        fun fromSection(section: ConfigurationSection): Generator {
            val generator = Generator(
                section.name,
                Material.matchMaterial(section.getString("drop_material"))
            )

            val block = section.getConfigurationSection("block")
            val world = Bukkit.getWorld(block.getString("world"))

            return generator.apply {
                this.block = world.getBlockAt(
                    block.getInt("x"),
                    block.getInt("y"),
                    block.getInt("z")
                )
            }
        }
    }
}